sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/javaCode.scala

```diff
@@ -148,14 +149,17 @@ trait Block extends JavaCode {
   }
 
   // Concatenates this block with other block.
-  def + (other: Block): Block
+  def + (other: Block): Block = other match {
+    case EmptyBlock => this
```

Comment:
what if `this` is an `EmptyBlock`? shall we a case also for it?

Reply:
An empty block + other empty block is an empty?

Reply:
sorry, probably the wrong line for the comment. I mean the case:  `EmptyBlock` + non-empty block. Shall we add a check and return other in that case?

Reply:
I had a idea to early expand `EmptyBlock` before. Now I commit it into. Then both concatenation and embedding cases, `EmptyBlock` won't be kept in arguments to code block.